File: Code/C++/row_wise_sum.cpp
/* ROW WISE SUM OF A 2D ARRAY :
For a given 2-D integer type array of size (N x M), find and print the sum of each of the row elements in a single line, separated by a single space.*/

#include <iostream>
using namespace std;

void row_wise_sum(int *arr,int r,int c)
{
    for (int i = 0; i < r ; i++)
    {
        int sum = 0;
        for (int j = 0; j < c ; j++)
        {
            sum += *((arr + i * c) + j);
        }
        cout << sum << " ";
    }
}

int main()
{
    int rows,cols;
    cout<<"Enter the number of rows and columns:";
    cin >> rows >> cols;
    int arr[rows][cols];

    cout<<"Enter the elements of the 2D array:\n";
    for(int i = 0; i < rows; i++)
    {
        for(int j = 0; j < cols; j++)
        {
            cin >> arr[i][j];
        }
    }

    cout<<"Row-wise Sum:"<<endl;
    row_wise_sum(&arr[0][0],rows,cols);
    return 0;
}
/* EXAMPLE 1:
Enter the number of rows and columns:3 4
Enter the elements of the 2D array:
1 2 3 4
0 -2 -1 3
-6 2 1 11
Row-wise Sum:
10 0 8

EXAMPLE 2:
Enter the number of rows and columns:3 3
Enter the elements of the 2D array:
1 2 3
6 4 2
4 5 7
Row-wise Sum:
6 12 16

Time Complexity - O(N * M)
Space Complexity - O(1) */

File: Code/Java/RowWise_Sum.java
/*
ROW WISE SUM OF A 2D ARRAY

Question: For a given two-dimensional integer array/list of size (N x M), 
find and print the sum of each of the row elements in a single line, 
separated by a single space.
 */
import java.util.*;
class solution{
//function to insert elements in 2d array
 public static int[][] take2DInput()
    {
          Scanner s=new Scanner(System.in);
          int n_rows=s.nextInt();  //number of rows 
           int n_cols=s.nextInt(); //number of columns
           
           if(n_rows==0)
           {
               return new int[0][0] ;
           }
           int mat[][]=new int[n_rows][n_cols];
           
           for(int row=0 ; row<n_rows ;row++)
           {
               for(int col=0 ; col<n_cols; col++)
               { 
                    int item=s.nextInt(); //element to insert in 2d array
                    mat[row][col]=item;
               
               }
           }
           return mat;
           
    }
	
//calculates row wise sum in 2d array	
 public static void rowWiseSum(int[][] mat) {
	//Your code goes here
        if(mat.length!=0){
         int rows=mat.length; //length os row
         int cols=mat[0].length;    //length of column  
         int sumit=0;
        
        for( int k=0;k<rows;k++){
            for(int j=0;j<cols;j++) {
             sumit+=mat[k][j]; //sum of row elements
                          
              }
              System.out.print(sumit+" ");
              sumit=0;
	}
   }
  }  


}

public class RowWise_Sum {        
    public static void main(String[] args) {
          //function call to insert elements in 2d array
          int[][] mat = solution.take2DInput();
          solution.rowWiseSum(mat);
          System.out.println();
          
    }

}
/*Sample Input 1:
4 2 
1 2 
3 4 
5 6 
7 8

output-->
3 7 11 15 

Input 2-->
2 5 
4 5 3 2 6 
7 5 3 8 9
Output-->
20 32 

Time Complexity - O(m*n) ,where m and n are no. of rows and columns resp.
Space Complexity - O(1) ,as no extra space is used except sum variable.
*/
